### PR TITLE
feat(medical_record): add initial feature

### PR DESCRIPTION
**Pull request by thisham**
## Overview
Medical record created by a doctor while consultation session has been ended. The medical record can only created, modified by doctors. This feature needs authenticated as doctor to access creation, and general authentication for reading.

## Testings
- [x] Create with doctor account
- [x] Prohibit create if not doctor
- [x] Get with patient NIK
- [x] Get by ID

### CODE REVIEW

File: src/app/medical_record/handlers/response/dto.go [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+		ICD10Description: domain.ICD10Description,
+		Suggestions:      domain.Suggestions,
+		Patient: PatientReference{
+			ID:        domain.Patient.ID,
+			Name:      domain.Patient.Name,
+			NIK:       domain.Patient.NIK,
+			Phone:     domain.Patient.Phone,
+			Address:   domain.Patient.Address,
+			Age:       domain.Patient.Age,
+			DOB:       domain.Patient.DOB,
```
Comment (dhichii) [inline src/app/medical_record/handlers/response/dto.go:65]: Please convert the value to string date

File: src/app/medical_record/repositories/record.go [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+		DoctorID:         domain.Doctor.ID,
+	}
+}
+
+func MapToBatchDomain(records []MedicalRecord) []medicalrecord.Domain {
+	var domains []medicalrecord.Domain
+
+	for _, record := range records {
+		domain := record.MapToDomain()
+		domain.Patient.Age = utils.CountIntervalByYearRoundDown(domain.Patient.DOB, domain.CreatedAt)
```
Comment (dhichii) [inline src/app/medical_record/repositories/record.go:111]: Why not use `time.Now()` instead of CreatedAt?

**Review by dhichii — CHANGES_REQUESTED** [review]

File: src/app/medical_record/repositories/record.go (hunk 2 above)
Comment (thisham) [reply src/app/medical_record/repositories/record.go:111]: If it using `time.Now()` the patient age of the stored medical records will be increase following the `now` year. But in storing a medical record, we need to know, "how old the patient when this record created?".

Comment (dhichii) [reply src/app/medical_record/repositories/record.go:111]: Oh, I see. Good job

File: libs/api-spec/api-spec.yml [hunk 3]
```diff
@@ -1296,6 +1462,136 @@ components:
...
+                      id: 2
+                      name: General
+            allOf:
+              - type: object
+                properties:
+                  data:
+                    type: array
+                    items:
+                      $ref: '#/components/schemas/MedicalRecord'
+              - $ref: '#/components/schemas/Meta_200OK'
```
[the remark this answers is not in the document]
Comment (thisham) [reply libs/api-spec/api-spec.yml]: Done at https://github.com/ALTERA-CAPSTON-41/backend/pull/46/commits/71f1e83d194c876e39790b16a840626c088a6c9b.

File: src/app/medical_record/repositories/record.go [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+	SIP    string           `gorm:"column:sip"`
+	Gender types.GenderEnum `gorm:"type:enum('MALE', 'FEMALE')"`
+}
+
+type ICDResponse struct {
+	Name        string   `json:"Name"`
+	ICDType     string   `json:"Type"`
+	Description string   `json:"Description"`
+	Diagnoses   []string `json:"Inclusions"`
+	Response    string   `json:"Response"`
```
[the remark this answers is not in the document]
Comment (thisham) [reply src/app/medical_record/repositories/record.go]: Done at https://github.com/ALTERA-CAPSTON-41/backend/pull/46/commits/bf7d94ba8c875913b3dcf9ad8e95aa50514e3a73.

**Review by dhichii — APPROVED** [review]
Nice job!

**Pull request merged**